From b1b8effc32b8bbf442674618284a03e4f6f09509 Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Wed, 19 Jun 2013 13:27:22 +0700
Subject: Dalvik: Correct assert statement in SSAWalkData.cpp

BZ: 117236

Assert statement incorrectly suggests that invvalid condition
should be evaluated to true.
The patch reverts the condition to be true if everything is good.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ib3adee76a459527b10fe88d0c337c36e2f0e2b22
Orig-MCG-Change-Id: I2bbfc0aaf1dda227ffb2cc2406f40e6c11d826f6
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/SSAWalkData.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/SSAWalkData.cpp b/vm/compiler/SSAWalkData.cpp
index 761ab46..64c3a2d 100644
--- a/vm/compiler/SSAWalkData.cpp
+++ b/vm/compiler/SSAWalkData.cpp
@@ -157,7 +157,8 @@ void SSAWalkData::handleNoDefinitions (void)
         SSARepresentation *ssaRep = mir->ssaRep;
 
         //Paranoid
-        assert (ssaRep == 0 || idx >= ssaRep->numUses);
+        assert (ssaRep != 0);
+        assert (0 <= idx && idx < ssaRep->numUses);
 
         //Get use value
         int value = ssaRep->uses[idx];
-- 
1.7.4.1

